### DISCUSSION TITLE
Can the job interval be scheduled to start after the completion of the previous run?

### DISCUSSION THREAD

**Discussion by FireSmack** [go-co-op/gocron#564] [Q&A] [answered]
I have a job set to run every 5 minutes:
```
s := gocron.NewScheduler(time.UTC)
job, err := s.Every(5).Minutes().Do(task)
if err != nil {
    log.Fatalf("Job %v, Error: %v", job, err)
}
```
I do not use `WaitForSchedule()` or `WaitForScheduleAll()`. 

I would like my task to always run 5 minutes after the previous iteration finishes, not every 5 minutes regardless of how long the previous run took.

Example:

Let's say the first execution of my job starts at 09:00:10 and ends at 09:00:20. The next execution will start at 9:05:10. However, is it possible to have the next execution start 5 minutes after the previous one _finishes_, so that the next one starts at 9:05:**20**? And then if that one finished at 09:05:35, I would want the following one to start at 09:10:35, not at 09:10:10. Is this possible?

**Comment by JohnRoesler** [comment, marked as answer]
Hey @FireSmack 

Not currently supported, but I can see how that would be a useful feature.

**Comment by husam-e** [comment]
I guess a trivial/naive way to do this would be to just have a `sleep(5 * time.Minute)` at the end of each iteration. This could also be done without a special scheduler library like this if it's the _only_ requirement (e.g. a for loop + sleep at the end, maybe with an injectable Clock or something to control for testing).

Would that work @FireSmack, or are there other reqs as well?

**Reply by husam-e** [reply]
A similar but slightly different feature would be to enforce a "minimum delay" between iterations. So if something is scheduled to run every second, but you want to make sure there's at least a 200ms "break" between one job and the next, if a job takes <= 800ms, it would tick as normal (run every second), but if it takes > 800 ms, it would always wait 200ms (maybe even with some jitter) before executing the next one. 

This would provide some breathing room and avoid overloading systems in scenarios where something is taking longer than expected or overexerting on resources.
